### PR TITLE
Create Intersector unit test from Intersector main

### PR DESCRIPTION
**Pull request by Darkyenus**
This creates an junit test in the style of `BitsTest` in `gdx/test`. Note that this test currently fails at line 28 (test_circle) due to bug in Circle.contains. Fix is in the followup PR. `intersectorTest` is currently not done yet, because I didn't have the time/tools/smarts yet to compute what should be the correct answer. So it wasn't included as a test yet.

### CODE REVIEW

File: gdx/test/com/badlogic/gdx/math/IntersectorTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		// Note, in stage the bottom and left sides are inclusive while the right and top sides are exclusive.
+	}
+
+	//@Test //No reason to test without asserts
+	public void intersectorTest () {
+		Plane plane = new Plane(new Vector3(1, 0, 0), 0);
+		SplitTriangle split = new SplitTriangle(3);
+		float[] fTriangle = {-10, 0, 10, -1, 0, 0, -10, 0, 10};
+		Intersector.splitTriangle(fTriangle, plane, split);
+		System.out.println(split);
```
Comment (saltares) [inline gdx/test/com/badlogic/gdx/math/IntersectorTest.java]: Could you please replace these println statements with asserts and make this a test with `@Test`?. Also remove comment before function.

**Comment by saltares** [comment]
Other than the comment I left, this looks good to me. Good stuff! Moar unit tests, MOAR!

**Comment by antag99** [comment]
The naming style used in `BitsTest` sucks. Use the standard java style, e.g. `testBoundingBox` rather than `test_boundingBox`.

**Comment by saltares** [comment]
@antag99 raises a good point.

File: gdx/test/com/badlogic/gdx/math/IntersectorTest.java (hunk 1 above)
Comment (Darkyenus) [inline gdx/test/com/badlogic/gdx/math/IntersectorTest.java]: I'd like to add proper asserts, but as I said before, I don't know what to assert for. I could assert that it outputs what it does now, but how can I know that it works now? If anybody knows what it should output, I can add it there.

**Comment by Darkyenus** [comment]
Agreed, I'll change it in both classes.

File: gdx/test/com/badlogic/gdx/math/IntersectorTest.java (hunk 1 above)
Comment (saltares) [inline gdx/test/com/badlogic/gdx/math/IntersectorTest.java]: The [method documentation](http://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/math/Intersector.html#splitTriangle-float:A-com.badlogic.gdx.math.Plane-com.badlogic.gdx.math.Intersector.SplitTriangle-) explains what the result should be.

**Comment by Tom-Ski** [comment]
Can one of the admins verify this patch?

**Comment by Tom-Ski** [comment]
Can one of the admins verify this patch?

**Comment by saltares** [comment]
Would you mind squashing everything into one commit? We can then verify the patch.

**Comment by Darkyenus** [comment]
I have changed the test names to have standard java method names and implemented the testSplitTriangle. And by chance, found a bug in Intersector.splitTriangle. See #3299.

**Comment by Darkyenus** [comment]
Squashed

**Comment by saltares** [comment]
Do we want to break the tests by merging this or should we wait for the fix for #3299 to be in? What's our policy on this?

**Comment by Darkyenus** [comment]
And this is it, everything should be working and tested properly now.

**Comment by Tom-Ski** [comment]
Cool stuff, any chance you can squash again? I will add the unit tests it to the gradle build so we can build them on the server.

**Comment by antag99** [comment]
Shouldn't the unit tests be moved into separate classes based on what classes they test? E.g. `testRectangle` goes in `RectangleTest` instead of `IntersectorTest`.

**Comment by Tom-Ski** [comment]
Yup

**Comment by Darkyenus** [comment]
Sure, I'll squash, I just wanted somebody to see what was I changing.

So, should I move the tests to their own classes now or will that wait for when there is more of them?

**Comment by Darkyenus** [comment]
Eh, I'll do it now with the squash.

**Comment by Tom-Ski** [comment]
We could move all the Shape2D tagged classes into a Shape2D test class.

**Comment by Darkyenus** [comment]
I have been thinking about that as well. I'll do that.

**Comment by Darkyenus** [comment]
Done.

**Comment by Tom-Ski** [comment]
Thanks!

**Comment by loogie** [comment]
Hello!

 

You have a new message, please read <http://husband.tv/attract.php?c>

 

loogie@basicelementdesign.com

**Pull request merged**